File: src/main/java/com/fatec/destino/repository/pacote/tag/TagRepository.java
package com.fatec.destino.repository.pacote.tag;

import com.fatec.destino.model.pacote.tag.Tag;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TagRepository extends JpaRepository<Tag, Long> {
    Optional<Tag> findByNome(String nome);
}

File: src/main/kotlin/com/fatec/destino/repository/pacote/tag/TagRepository.kt
package com.fatec.destino.repository.pacote.tag

import com.fatec.destino.model.pacote.tag.Tag
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TagRepository : JpaRepository<Tag, Long> {
    fun findByNome(nome: String): Tag?
    fun findByNomeIn(nomes: List<String>): Set<Tag>
}
